src/main/java/org/radarbase/appserver/service/protocol/GithubClient.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ *  *
+ *  *  * Copyright 2018 King's College London
+ *  *  *
+ *  *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  *  * you may not use this file except in compliance with the License.
+ *  *  * You may obtain a copy of the License at
+ *  *  *
+ *  *  *   http://www.apache.org/licenses/LICENSE-2.0
+ *  *  *
+ *  *  * Unless required by applicable law or agreed to in writing, software
+ *  *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *  * See the License for the specific language governing permissions and
+ *  *  * limitations under the License.
+ *  *  *
+ *  *
+ *
+ */
+
+package org.radarbase.appserver.service.protocol;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+@Slf4j
+@Component
+@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
+public class GithubClient {
+
+    private static final String GITHUB_API_URI = "api.github.com";
+    private static final String GITHUB_API_ACCEPT_HEADER = "application/vnd.github.v3+json";
+    private final transient ObjectMapper objectMapper;
+    private final transient HttpClient client;
+
+    @Value("${security.github.client.token}")
+    private transient String githubToken;
+
+    @SneakyThrows
+    @Autowired
+    public GithubClient(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+        client = HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(10)).build();
+    }
+
+    private static boolean isSuccessfulResponse(HttpResponse response) {
+        return response.statusCode() >= 200 && response.statusCode() < 300;
+    }
+
+    public String getGithubContent(String url) throws IOException, InterruptedException {
+        URI uri = URI.create(url);
+        if (!this.isValidGithubUri(uri)) {
+            throw new IOException("Invalid Github url.");
```

Comment:
a better would be `MalformedURLException`. It extends `IOException` so no need to change any other code.